lib/notifications/notification_server.ex

```diff
@@ -146,8 +147,14 @@ defmodule Notifications.NotificationServer do
 
   defp broadcast(notifications, registry_key) do
     Registry.dispatch(Notifications.Supervisor.registry_name(), registry_key, fn entries ->
-      Enum.each(entries, fn {pid, _} ->
-        send(pid, {:notifications, notifications})
+      Enum.each(notifications, fn notification ->
+        usernames = User.usernames_for_route_ids(notification.route_ids)
+
+        Enum.each(entries, fn {pid, username} ->
+          if Enum.member?(usernames, username) do
+            send(pid, {:notifications, notifications})
```

Comment:
```suggestion
            send(pid, {:notification, notification})
```
Now that you're looping over the notifications one by one, I think this needs to be singular.